src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs

```diff
@@ -999,6 +999,12 @@ private static ulong CurlReceiveHeadersCallback(IntPtr buffer, ulong size, ulong
                                     // our redirect limit, etc.), this will have been unnecessary work in reconfiguring the easy handle, but 
                                     // nothing incorrect, as we'll tear down the handle once the request finishes, anyway, and all of the configuration
                                     // we're doing is about initiating a new request.
+                                    if ((int)response.StatusCode >= 301 && (int)response.StatusCode <= 303)
```

Comment:
why use (int) here?  Why not just use the HttpStatusCode enum type directly?

Reply:
I based this on the lines just above this one:
https://github.com/dotnet/corefx/blob/444e6bb7fd274be680fb3bf4b22258821e77ae2c/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs#L991-L993
But I can always adjust to use `HttpStatusCode`. That would definitely be more clear.

Reply:
I don't think that is a best-practice pattern either.  The StatusCode property already has a type associated with it (HttpStatusCode). So, the best thing is just leave it as that type and use the enum values to compare.

Reply:
Sounds good. I'll adjust my change and update the lines above as well.

Reply:
>update the lines above as well.

The lines above which are testing for multiple values in a range might be ok to use `int` casts. But for checking individual values leaving it as HttpStatusCode is best.

Reply:
My code is also checking for a range of values. I could change it to check each of the three relevant status codes, but the current code makes fewer comparisons.